Modelica compiler error use of undeclared identifier

Hi,
### Description

I get a compiler error with the model below. The model should generate a clock with random intervals (a poisson process in physics)
Instead I get 

> test_Poisson_15syn.c:39:61: error: use of undeclared identifier '_omcQ_24CLKPRE'
>       data->simulationInfo->baseClocks[base_idx].interval = _omcQ_24CLKPRE._Poisson_source1._NextInterval;
>                                                             ^
> 1 error generated.
> mingw32-make: *** [<builtin>: test_Poisson_15syn.o] Error 1
> mingw32-make: *** Waiting for unfinished jobs....
> mingw32-make: Leaving directory 'C:/Tmp/OPENMO~1/OMEdit/SR9CA2~1.TES'
> Compilation process failed. Exited with code 0x00000002.
> 
Needless to say that the name _omcQ_24CLKPRE is not a model variable, but something made up by the modelica compiler.

### Steps to Reproduce
Please paste the model below in openmodelica and try to simulate. 
```
model test_Poisson

block PoissonProcess "Generate fixed magnitude events with exponentially distributed interval"
  import generator = Modelica.Math.Random.Generators.Xorshift128plus;
  import Modelica.Math.Random.Utilities.automaticLocalSeed;
  import distribution = Modelica.Math.Distributions.Uniform.quantile;
  extends Modelica.Blocks.Interfaces.SO;

  discrete Real NextInterval(start=1e-3) "Waiting time until next event";
  Clock Clock1 = Clock(NextInterval);
  parameter Real lambda=1440 "Event rate (#events per second)";
  Integer eventcounter(start=0) "Verification only (did we get the correct expected value?)";

    // Declare state and random number variables
  Integer state[generator.nState] "Internal state of random number generator";
  discrete Real r_raw "Uniform random number in the range (0,1]";
initial equation
   pre(state) = generator.initialState(0, 1);

equation
  when Clock1 then
    (r_raw, state) = generator.random(pre(state));
    NextInterval=-log(distribution(r_raw, 0, 1))/lambda; // Conversion from wikipedia exponential distribution https://en.wikipedia.org/wiki/Exponential_distribution
    y=1;
    eventcounter=previous(eventcounter)+1;
  end when;

  annotation (Icon(graphics={Line(points={{85,-40},{-85,-40}}, color={28,108,200}), Line(points={{-64,40},{-64,-40}}, color={28,108,200}), Line(points={{-54,40},{-54,-40}}, color={28,108,200}), Line(points={{-15,40},{-15,-40}}, color={28,108,200}), Line(points={{50,40},{50,-40}}, color={28,108,200}), Line(points={{70,40},{70,-40}}, color={28,108,200})}));
end PoissonProcess;

  PoissonProcess Poisson_source1 annotation (Placement(transformation(extent={{-12,30},{8,50}})));
  inner Modelica.Blocks.Noise.GlobalSeed globalSeed(useAutomaticSeed=false) annotation (Placement(transformation(extent={{-80,52},{-60,72}})));
  annotation (Icon(coordinateSystem(preserveAspectRatio=false)), Diagram(coordinateSystem(preserveAspectRatio=false)));
end test_Poisson;

```

### Expected Behavior

Succesfull compilation, as in dymola

### Version and OS
v1.19.0-dev-504-g8291efd717 (64-bit) on windows 10

--- a/OMCompiler/Compiler/BackEnd/SynchronousFeatures.mo
+++ b/OMCompiler/Compiler/BackEnd/SynchronousFeatures.mo
@@ -2302,9 +2302,8 @@ algorithm
     then (DAE.REAL_CLOCK(e), eqs, vars, cnt);
 
     case DAE.RATIONAL_CLOCK(e, i) equation
-      // AHeu: Surpressing substClockExp to get rid of previous in intervalCounter
-      //(e, eqs, vars, cnt) = substClockExp(e, inNewEqs, inNewVars, inCnt, inShared);
-    then (DAE.RATIONAL_CLOCK(e, i), inNewEqs, inNewVars, inCnt);
+      (e, eqs, vars, cnt) = substClockExp(e, inNewEqs, inNewVars, inCnt, inShared);
+    then (DAE.RATIONAL_CLOCK(e, i), eqs, vars, cnt);
 
     else (inClk, inNewEqs, inNewVars, inCnt);
   end match;
@@ -2362,10 +2361,6 @@ algorithm
     outCnt := inCnt;
   else
     ({outExp}, outNewEqs, outNewVars, outCnt) := substExp({inExp}, inNewEqs, inNewVars, inCnt);
-    outExp := match outExp
-      case DAE.CREF(_, ty) then Expression.makePureBuiltinCall("previous", {outExp}, ty);
-      else outExp;
-    end match;
   end if;
 end substClockExp;
 

--- a/OMCompiler/Compiler/Template/CodegenC.tpl
+++ b/OMCompiler/Compiler/Template/CodegenC.tpl
@@ -357,6 +357,11 @@ template baseClockInit(ClockKind baseClock, Integer baseClockIdx, list<SubPartit
     else
       '0'
   let interval = match baseClock
+    case REAL_CLOCK() then
+      if isConst(interval) then
+        daeExp(interval, contextOther, &preExp, &varDecls, &auxFunction)
+      else
+        '-1 /* Interval set in _updateSynchronous */'
     case INFERRED_CLOCK() then
       '1'
     else
@@ -471,11 +476,16 @@ template updatePartition(Integer i, DAE.ClockKind baseClock, Text &varDecls, Tex
         data->simulationInfo->baseClocks[base_idx].interval = DIVISION((modelica_real)data->simulationInfo->baseClocks[base_idx].intervalCounter, (modelica_real)data->simulationInfo->baseClocks[base_idx].resolution, "intervalCounter/resolution");
         >>
     case REAL_CLOCK() then
-      let interval = daeExp(getClockInterval(baseClock), contextOther, &preExp, &varDecls, &auxFunction)
-      <<
-      <%preExp%>
-      data->simulationInfo->baseClocks[base_idx].interval = <%interval%>;
-      >>
+      if isConst(interval) then
+        <<
+        /* Nothing to do */
+        >>
+      else
+        let interval_ = daeExp(interval, contextOther, &preExp, &varDecls, &auxFunction)
+        <<
+        <%preExp%>
+        data->simulationInfo->baseClocks[base_idx].interval = <%interval_%>;
+        >>
     case INFERRED_CLOCK()
     case SOLVER_CLOCK()
     case EVENT_CLOCK() then

